src/agent/src/rpc.rs

```diff
@@ -1722,6 +1747,150 @@ fn load_kernel_module(module: &protocols::agent::KernelModule) -> Result<()> {
     }
 }
 
+fn pull_image_from_registry(image: &str, src_creds: &str) -> Result<()> {
+    let source_image = format!("{}{}", "docker://", image);
+
+    // TODO LATER derive directory from image registry/namespace/repository/tag to make this more dynamic
+    // Define the target tranport and path for the manifest image, with signature
```

Comment:
Couldn't we just prefix it with the container ID instead, and pave the way for a full pod images pull?

```Rust
let target_path_manifest = format!("dir://{}/image_manifest", cid);
```